src/dom.ts

```diff
@@ -269,3 +269,67 @@ export function compareElementArea(a: HTMLElement, b: HTMLElement): number {
 
     return aArea - bArea
 }
+
+/** Computes a unique CSS selector for the current HTMLElement.
+ *  Currently buggy, returns a unique selector but doesn't always correspond
+ *  to the selector you should use for functions such as querySelectorAll().
+ */
+export function getSelector(elem: HTMLElement) {
+   if (elem === null || elem === undefined)
+      return undefined
+
+   if (elem == document.body || !elem.parentNode)
+      return "body"
+
+   if (elem.id)
+      return "#" + elem.id
+
+   let parent = elem.parentNode as HTMLElement
+   let n = [].indexOf.call(parent.childNodes, elem) + 1
+   return getSelector(parent) + " :nth-child(" + n + ")"
+}
+
+export const hintworthy_js_elems = {}
```

Comment:
This is an object because I needed it to cache the correspondence between selectors and elements for a previous attempt and it stayed. An array of element would indeed be better, thanks for spotting that.

It seems I can get stuck in quite awful local optimums when programming!